src/components/gitProvider/GitProvider.tsx

```diff
@@ -133,19 +234,37 @@ function GitForm({ id = null, name = "", active = false, url = "", authMode = nu
             }
         }
 
+        let gitHostId = gitHost.value.value;
+        if (gitHost.value.__isNew__) {
+            let gitHostPayload = {
+                name: gitHost.value.value,
+                active:true
+            }
+            try {
+                const { result } = await saveGitHost(gitHostPayload);
+                getHostList();
+                gitHostId = result;
+            } catch (error) {
+                showError(error)
+            }
+        }
+
+        console.log(hostListOption)
         let payload = {
             id: id || 0,
             name: state.name.value,
+            gitHostId: gitHostId,
             url: state.url.value,
             authMode: state.auth.value,
             active,
             ...(state.auth.value === 'USERNAME_PASSWORD' ? { username: customState.username.value, password: customState.password.value } : {}),
             ...(state.auth.value === 'ACCESS_TOKEN' ? { accessToken: customState.accessToken.value } : {})
         }
+
         const api = id ? updateGitProviderConfig : saveGitProviderConfig
         try {
             setLoading(true)
-            const { result } = await api(payload, id);
+            await api(payload, id);
             await reload();
```

Comment:
[result1, result2] = Promise.all([async1(), async2()]); can be used to combine line no 267 and 268